Orange/widgets/unsupervised/owhierarchicalclustering.py

```diff
@@ -410,6 +415,13 @@ def _set_displayed_root(self, root):
         )
         self._main_graphics.layout().activate()
 
+    def _set_input_summary(self, matrix):
+        summary, details = self.info.NoInput, ""
+        if matrix is not None and matrix.row_items:
+            summary = len(matrix.row_items)
```

Comment:
DistMatrix does not necessary have row_items. I'd provide only `summary` (`len(matrix)`) and no `detail`.